src/event.ts

```diff
@@ -33,21 +33,29 @@ export class PropertyDetail implements DataLayerDetail {
 /**
  * Defines CustomEvent types. Types will be prefixed with a DLO namespace.
  * See https://developer.mozilla.org/en-US/docs/Web/API/Event/type
+ * @param source from the rule monitoring the data layer
```

Comment:
just making sure I understand the change, `source` is the entire string of source in the rule, and `path` is the object without the "selection"? Sorry I'm still a bit fuzzy on the concepts here.
For example, if the rule has a `source: s[^(eVar1)]`, this `source` would be `s[^(eVar1)]` and path is `s`?

Reply:
You've got that right. `source` is taken directly from the `source` value configured on the rule, whereas `path` is the object being monitored. `source` and `path` can be the same, or they can be different. Some examples:

```
source: nestedData.child, path: nestedData.child
source: s[^(eVar1)], path: s
```

`path` is the object that gets shims defined for its properties and functions to trigger rules on property value changes and function calls.

Previously, each property or function could only be shimmed once, and each shim could only trigger one event path. So, if multiple rules applied to the same `path` object (like `s`), all those rules had to be triggered on property value change or function call even if a selector didn't care about the property that changed or the function that was called.

Reply:
Okay I'm still wrapping my head around it, so each path/object will still be shimmed once, but any changes get monitored, it will emit an even per source, per path. And then only the rules that matches both will care about the event and process the data? 
^ That SGTM! (with my limited DLO knowledge 😅) Only thing I wonder is how much more "chatty" would this make dlo's custom events, but since rules are manually added, it's probably not a big deal?

Reply:
> so each path/object will still be shimmed once, but any changes get monitored, it will emit an even per source, per path. And then only the rules that matches both will care about the event and process the data?

That all sounds correct to me. In terms of chattiness, DLO will raise more browser events (`window.dispatch`) when multiple rules are monitoring the same path. Since this only affects cases where multiple rules monitor the same path and rules are explicitly defined (via OOTB rulesets or via custom rules) -- and since browser events are typically high volume (mouse movement, key down, key up, click, etc.) -- I'm expecting the impact to be minimal.

In contrast, this change will reduce `FS.event` chattiness for rules monitoring the same path where the data doesn't match a rule's selector pattern.